transcom/mymove
# B-20803 duplicate fee MM - INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A984986)

## Summary

Multiple move management service items were being created in the case of a user double clicking the "Approve and send" button as the TOO. This solution uses the debounce method. I used the debounce from lodash since that was a preexisting library in our code. I used this as the guide: https://dev.to/stokemasterjack/prevent-double-click-dups-in-react-5cc6.

To test:
1. As a services counselor, counsel a move
2. As a TOO, find that move and get to the point of seeing the "Approve and send" button.
3. Click it multiple times.
4. On the MTO page, check to make sure that only one Move management service item was created.

Before (screenshot borrowed from Agility attachment):
![image](https://github.com/user-attachments/assets/1150f09a-896b-4825-8585-469eb9130ffd)

After:
![image](https://github.com/user-attachments/assets/60275eab-ceaa-413a-92a4-6ab4e31d7798)
